### PR TITLE
Add warnings for changes in execdrivers

### PR DESCRIPTION
**Pull request by tonistiigi**
Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

File: github/comments.go [hunk 1]
```diff
@@ -36,6 +36,12 @@ Ammending updates the existing PR. You **DO NOT** need to open a new one.
 	return g.addUniqueComment(repo, strconv.Itoa(pr.Number), comment, "sign your commits", content)
 }
 
+func (g GitHub) addExecdriverDeprecationComment(repo octokat.Repo, pr *PullRequest, content *PullRequestContent) error {
+	comment := `Please note that concept of execdrivers is being replaced with OCI compliant binaries executed through containerd. There is an ongoing effort for switching to containerd in #20662 . Please consider porting the changes in your PR to this branch instead.`
+
+	return g.addUniqueComment(repo, strconv.Itoa(pr.Number), comment, "execdriver deprecation", content)
```
Comment (thaJeztah) [inline github/comments.go]: you need to change the "execdriver deprecation" string to something unique that's part of the comment you're adding, as it is used to check if the comment is already added https://github.com/docker/leeroy/blob/96bf4b85ae2ea0d391bdf0d87de51d6ad85eb12c/github/pull_request.go#L138

File: github/dco.go [hunk 2]
```diff
@@ -97,6 +97,14 @@ func (g GitHub) DcoVerified(pr *PullRequest) (bool, error) {
 	return verified, nil
 }
 
+// CheckExecdriver checks if the pull request changes something in execdrivers
+func (g GitHub) CheckExecdriver(pr *PullRequest) error {
+	if pr.Execdriver() {
+		return g.addExecdriverDeprecationComment(pr.Repo, pr, pr.Content)
+	}
+	return g.removeComment(pr.Repo, "execdriver deprecation", pr.Content)
```
Comment (thaJeztah) [inline github/dco.go]: same here

File: github/comments.go (hunk 1 above)
Comment (tonistiigi) [inline github/comments.go]: PTAL

**Comment by thaJeztah** [comment]
LGTM

**Comment by calavera** [comment]
LGTM

**Pull request merged**